modules/core/src/main/java/org/apache/ignite/internal/cache/query/index/IndexQueryProcessor.java

```diff
@@ -125,6 +131,38 @@ public <K, V> IndexQueryResult<K, V> queryLocal(
 
                 IgniteBiTuple<K, V> row = currVal;
 
+                if (cctx.events().isRecordable(EVT_CACHE_QUERY_OBJECT_READ) &&
+                    cctx.gridEvents().hasListener(EVT_CACHE_QUERY_OBJECT_READ)) {
+
+                    final K key = row.getKey();
+                    final V val = row.getValue();
+
+                    CacheObjectContext objCtx = cctx.cacheObjectContext();
+
+                    K key0 = (K)CacheObjectUtils.unwrapBinaryIfNeeded(objCtx, key, keepBinary, false, null);
```

Comment:
This code duplicates code in the `onHasNext` method.

Reply:
Made this section of code more concise by using the `unwrap()` method.